inc/wp-components/classes/class-head.php

```diff
@@ -207,4 +138,287 @@ public function add_tag( $tag, $attributes = [] ) : self {
 		return $this->append_child( $component );
 	}
 
+	/**
+	 * Hook into post being set.
+	 *
+	 * @return self
+	 */
+	public function query_has_set() : self {
+		$this->set_title( $this->get_the_head_title() . $this->get_trailing_title() );
+		return $this;
+	}
+
+	/**
+	 * Hook into post being set.
+	 *
+	 * @return self
+	 */
+	public function post_has_set() : self {
+
+		$this->set_title( $this->get_meta_title() . $this->get_trailing_title() );
+		$this->set_open_graph_meta();
+
+		return $this;
+	}
+
+	/**
+	 * Get the head title based on the query set.
+	 *
+	 * @return string
+	 */
+	public function get_the_head_title() : string {
+		switch ( true ) {
+			// Search results.
+			case $this->query->is_search():
+				return sprintf(
+					/* translators: search term */
+					__( 'Search results: %s', 'wp-components' ),
+					$this->query->get( 's' )
+				);
+
+			// Author archive.
+			case $this->query->is_author():
+				$this->set_author( $this->query->get( 'author_name' ) );
+				return sprintf(
+					/* translators: author display name */
+					__( 'Articles by %s', 'wp-components' ),
+					$this->get_author_display_name()
+				);
+
+			// Term archives.
+			case $this->query->is_category():
+			case $this->query->is_tag():
+			case $this->query->is_tax():
+				$this->set_term( $this->query->get_queried_object() );
+				return $this->wp_term_get_name();
+
+			// Generic 404.
+			case $this->query->is_404():
+				return __( '404 - Page not found', 'wp-components' );
+
+			// Post type archives.
+			case $this->query->is_post_type_archive():
+				$post_type   = $this->query->get( 'post_type' );
+				$post_object = get_post_type_object( $post_type );
+				return $post_object->label;
+		}
+	}
```

Comment:
This function should return *something* if none of the cases match, so you either need a `default` case in your `switch` or a default return statement at the bottom of the function.

Reply:
Right, I made a mental note to do that and forgot. Thanks.